libremesh.mk

```diff
@@ -0,0 +1,63 @@
+include $(TOPDIR)/rules.mk
+
+LIME_NAME=$(notdir ${CURDIR})
+
+PKG_NAME?=$(LIME_NAME)
+PKG_MAINTAINER?=$(LIME_MAINTAINER)
+
+# from https://github.com/openwrt/luci/blob/master/luci.mk
+PKG_VERSION?=$(if $(DUMP),x,$(strip $(shell \
+	if svn info >/dev/null 2>/dev/null; then \
+		revision="svn-r$$(LC_ALL=C svn info | sed -ne 's/^Revision: //p')"; \
+	elif git log -1 >/dev/null 2>/dev/null; then \
+		revision="svn-r$$(LC_ALL=C git log -1 | sed -ne 's/.*git-svn-id: .*@\([0-9]\+\) .*/\1/p')"; \
+		if [ "$$revision" = "svn-r" ]; then \
+			set -- $$(git log -1 --format="%ct %h" --abbrev=7); \
+			secs="$$(($$1 % 86400))"; \
+			yday="$$(date --utc --date="@$$1" "+%y.%j")"; \
+			revision="$$(printf 'git-%s.%05d-%s' "$$yday" "$$secs" "$$2")"; \
+		fi; \
+	else \
+		revision="unknown"; \
+	fi; \
+	echo "$$revision" \
+)))
+PKG_RELEASE?=1
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/$(PKG_NAME)
+  SECTION:=lime
+  CATEGORY:=LibreMesh
+  TITLE:=$(LIME_TITLE)
+  DEPENDS:=+lime-system $(LIME_DEPENDS)
```

Comment:
Not every package depends on `lime-system` as an example `shared-state` doesn't, but with this change it would install lime-system too when it is not needed

Reply:
Good point, what would you suggest? Should everyone just add the package in the *LiMe* Makefile?

Not relying on `lime-system` sounds like upstream material to me. What about adding those packages to the official package repository?

Reply:
I agree that there's plenty of stuff which should be upstreamed (bandwidth-test, batman-adv-auto-gw-mode, bmx6-auto-gw-mode, bmx7-auto-gw-bw-mode, bmx7-auto-gw-mode, bmx7-mdns, check-date-http, hotplug-initd-services, lua-argparse, luci-app-batman-adv, luci-app-openairview, luci-mod-status, pirania-app, pirania, shared-state-babeld_hosts, shared-state-bat_hosts, shared-state-dnsmasq_hosts, shared-state-dnsmasq_leases, shared-state, shared-state-persist, shared-state-pirania, soporteremoto, tinc) but in the meanwhile I suggest to remove lime-system from the libremesh.mk profile and to add it to the single Makefiles.

Reply:
Sorry, I changed my mind: it's better to leave the lime-system dependency inside libremesh.mk and use libremesh.mk only for LibreMesh specific packages.
In this PR the following Makefiles are being edited:

auto-usb-wwan - NOT LibreMesh specific
bmx7-auto-gw-mode - NOT LibreMesh specific
lime-proto-babeld - LibreMesh specific
lime-proto-batadv - LibreMesh specific
lime-proto-bmx6 - LibreMesh specific
lime-proto-bmx7 - LibreMesh scecific
lime-proto-wan - LibreMesh specific
lime-smart-wifi - LibreMesh specific

@aparcar could you remove auto-usb-wwan and bmx7-auto-gw-mode Makefiles from this PR?

Reply:
Removed